src/main/java/pingis/services/DataImporter.java

```diff
@@ -141,14 +143,14 @@ public void generateEntities() {
 
     for (int i = 0; i < challenges.length(); i++) {
       JSONObject challengeObject = challenges.getJSONObject(i);
-      JSONArray tasks = challengeObject.getJSONArray("tasks");
-
       Challenge challenge = new Challenge(challengeObject.getString("name"),
           users.get(challengeObject.getString("author")),
           challengeObject.getString("desc"));
+      challenge.setType(ChallengeType.getType(challengeObject.getString("type")));
+      challenge.setRealm(Realm.getRealm(challengeObject.getString("realm")));
```

Comment:
Consider adding a constructor to `Challenge` and deserializing the JSON with Jackson (see usage of `ObjectMapper` in some classes) instead of manually getting all the attributes.